src/Front/Ast.hs

```diff
@@ -115,6 +125,18 @@ collectApp t               = (t, [])
 instance Pretty Program where
   pretty (Program defs) = vsep (intersperse emptyDoc $ map pretty defs)
 
+instance Pretty TopDef where
+  pretty (TopDef d) = pretty d
+  pretty (TopClass (tcid, tvid, ms)) =
+    pretty "class" <+> pretty tcid <+> pretty tvid <+> braces
+      (hsep $ punctuate semi $ map
+        (\(vid, typann) -> pretty vid <+> pretty typann)
```

Comment:
How about `bimap pretty pretty` instead?

Reply:
`bimap pretty pretty` gives `[(Doc, Doc)]`, while we need `[Doc]` here. So I think the lambda is necessary to collapse the tuple.

Reply:
`bimap pretty pretty` gives `[(Doc, Doc)]`, while we need `[Doc]` here. So I think the lambda is necessary to collapse the tuple.